### ISSUE TITLE
Doesn't compile with 6.9.5

### ISSUE THREAD

**Issue by crooney** [purescript/purescript-lists#24]
hello,

This afternoon I cleared my bower cache and updated it and now my builds fail in purescript-lists.  The package I'm trying to use that includes lists is purescript-simple-dom, if that matters. I took a gander at what the problem could be, but with 4 days of purescript under my belt, I'm not much use.

I tried reverting to 6.0, but the problem persists.

Transcript follows. If you need any further info from me, I am at your service.

Thanks
```
[ptest]$ psc --version                                                        proj/git/master !+
0.6.9.5
[]$ mkdir ptest                                                               proj/git/master !+
[]$ cd ptest                                                                  proj/git/master !+
[ptest]$ bower cache clean                                                    proj/git/master !+
bower deleted       Cached package purescript-math: /home/crooney/.cache/bower/packages/06794e306037a75c5b8e4d97f6427f43/0.1.1
.
. [excised]
.
bower deleted       Cached package purescript-refs: /home/crooney/.cache/bower/packages/be59257972dfcae16d2a6aafd3401b82/0.1.3
[ptest]$ bower install purescript-lists                                       proj/git/master !+
bower not-cached    git://github.com/paf31/purescript-lists.git#*
bower resolve       git://github.com/paf31/purescript-lists.git#*
bower download      https://github.com/paf31/purescript-lists/archive/v0.6.1.tar.gz
bower extract       purescript-lists#* archive.tar.gz
bower invalid-meta  purescript-lists is missing "main" entry in bower.json
bower resolved      git://github.com/paf31/purescript-lists.git#0.6.1
.
. [excised]
.
[ptest]$ pulp init                                                            proj/git/master !+
* Generating project skeleton in /home/crooney/proj/ptest
[ptest]$ pulp build                                                           proj/git/master !+
* Building project in /home/crooney/proj/ptest
Compiling Prelude
.
. [excised]
.
Compiling Control.Monad.ListT
Error in module Control.Monad.ListT:
Error in value declaration applyListT:
Error at bower_components/purescript-lists/src/Control/Monad/ListT.purs line 220, column 15 - line 220, column 18:
  Cannot unify type
    Control.Monad.ListT.ListT
  with type
    Prim.Function
See https://github.com/purescript/purescript/wiki/Error-Code-ErrorUnifyingTypes for more information, or to contribute content related to this error.

* ERROR: Subcommand terminated with error code 1

```

**Comment by garyb** [comment]
Hi @crooney, sorry we didn't get to this earlier. Did you manage to find a workaround in the end, or are you still having issues?

The version of `purescript-lists` in `master` is intended for a new compiler release, and it looks from your log here like there was no dependency version specified for the library (`"git://github.com/paf31/purescript-lists.git#*"`) - you could try installing the latest non-RC version explicitly: `bower install purescript-lists#0.6.2`

**Comment by crooney** [comment]
Don't worry about the delay at all. I was just checking out purescript at the time and fooling around. I did try an earlier version (6.3?) before creating the issue. Apparently I should have gone back one further.

At the time I was using a node-installed purescript. I've since switched to the nix version, which also reports 0.6.9.5. I'd kind of forgotten about this, but I just tried the same steps which resulted in the error, and it builds fine now, whether because of the differing node install or subsequent changes to purescript-lists, I couldn't say.

In any event thanks for your kind attention, help and the project itself.
